File: Stack/stack.py
class Stack:
    def __init__(self):
        self.arr = []
    
    def is_empty(self):
        return len(self.arr) == 0
    
    def push(self, data):
        self.arr.append(data)
    
    def pop(self):
        if self.is_empty():
            raise Exception("Stack Underflow!")
        return self.arr.pop()
    
    def peek(self):
        if self.is_empty():
            raise Exception("Stack is empty!")
        return self.arr[-1]

stack = Stack()
stack.push(1)
stack.push(2)
stack.push(3)
print("Top element:", stack.peek())
print("Popped element:", stack.pop())
print("Top element:", stack.peek())

File: Stack/stack.cpp
#include <iostream>
using namespace std;

#define MAX_SIZE 100

class Stack {
private:
    int arr[MAX_SIZE];
    int top;

public:
    Stack() {
        top = -1;
    }

    bool isEmpty() {
        return top == -1;
    }

    bool isFull() {
        return top == MAX_SIZE - 1;
    }

    void push(int data) {
        if (isFull()) {
            cout << "Stack Overflow!" << endl;
            return;
        }
        arr[++top] = data;
    }

    int pop() {
        if (isEmpty()) {
            cout << "Stack Underflow!" << endl;
            return -1;
        }
        return arr[top--];
    }

    int peek() {
        if (isEmpty()) {
            cout << "Stack is empty!" << endl;
            return -1;
        }
        return arr[top];
    }
};

int main() {
    Stack stack;
    stack.push(1);
    stack.push(2);
    stack.push(3);
    cout << "Top element: " << stack.peek() << endl;
    cout << "Popped element: " << stack.pop() << endl;
    cout << "Top element: " << stack.peek() << endl;
    return 0;
}
